trillian/ctfe/cert_checker.go

```diff
@@ -118,25 +118,44 @@ func ValidateChain(rawChain [][]byte, validationOpts CertValidationOpts) ([]*x50
 	// so that it doesn't affect EKU validity calculations.  In particular, if
 	// the pre-issuer has just the CT EKU, then it should act as if it has an
 	// empty set of EKUs (and so allow any usage in the leaf).
+	havePreissuer := false
 	var originalEKUs []x509.ExtKeyUsage
 	if len(chain) > 1 {
 		for i, eku := range chain[1].ExtKeyUsage {
 			if eku == x509.ExtKeyUsageCertificateTransparency {
 				originalEKUs = chain[1].ExtKeyUsage
 				chain[1].ExtKeyUsage = append(chain[1].ExtKeyUsage[:i], chain[1].ExtKeyUsage[i+1:]...)
+				havePreissuer = true
 				break
 			}
 		}
 	}
 
+	if havePreissuer {
```

Comment:
What would the cost be of making a new modified chain? Then it wouldn't have to be undone.

Reply:
The runtime cost isn't a worry, it's the swapping things around that would be fiddly and error-prone because everything uses pointers to the same set of `x509.Certificates`.  The call to `chain[0].Verify()` generates the returned chain, but that in turn relies on the intermediate pool that's populated from the chain.  If we had two copies of the chain (modified and unmodified) we'd have to do index mapping from one to the other to rebuild the analogous chain of unmodified certs.